source/Octopus.Client/SpaceContext.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Octopus.Client.Exceptions;
+using Octopus.Client.Util;
+
+namespace Octopus.Client
+{
+    public class SpaceContext
+    {
+        public static SpaceContext SpecificSpace(string spaceId) => new SpaceContext(new [] {spaceId}, false);
+        public static SpaceContext SpecificSpaceAndSystem(string spaceId) => new SpaceContext(new []{spaceId}, true);
+        public static SpaceContext SystemOnly() => new SpaceContext(new string[0], true);
+
+        internal SpaceContext(IReadOnlyCollection<string> spaceIds, bool includeSystem)
+        {
+            if (spaceIds.Count == 0 && !includeSystem)
+                throw new ArgumentException("At least 1 spaceId is required when includeSystem is set to false");
```

Comment:
should this also throw `MismatchSpaceContextException`?

Reply:
I feel `ArgumentException` is a bit more appropriate as it does not mismatch with anything yet, it is some invalid arguments being passed in that cause the `SapceContext` become not usable, what do you think? @NickJosevski

Reply:
I see yea that's fine then.